src/utils/decode-media-data.ts

```diff
@@ -1,14 +1,84 @@
-import { title } from 'case'
+import { title as titleCase } from 'case'
 
 import { decodeData as decodeString } from '~/utils/decode-data'
 import type { Media, Tag } from '~/models/media'
+import type { MediaType } from '~/constants/media'
+import { AUDIO, IMAGE, MODEL_3D, VIDEO } from '~/constants/media'
 
 /**
  * This interface is a subset of `Media` that types dictionaries sent by the API
  * being decoded in the `decodeMediaData` function.
  */
-interface ApiMedia extends Omit<Media, 'frontendMediaType' | 'title'> {
+interface ApiMedia
+  extends Omit<Media, 'frontendMediaType' | 'title' | 'originalTitle'> {
   title?: string
+  originalTitle?: string
+}
+
+const mediaTypeExtensions: Record<MediaType, string[]> = {
+  [IMAGE]: ['jpg', 'jpeg', 'png', 'gif', 'svg'],
+  [AUDIO]: ['mp3', 'wav', 'ogg', 'flac', 'aac', 'aiff'],
+  [VIDEO]: ['mp4', 'webm', 'mkv', 'avi', 'mov', 'wmv', 'flv', 'mpg', 'mpeg'],
+  [MODEL_3D]: ['fbx', 'obj', 'stl', 'dae', '3ds', 'blend', 'max', 'obj', 'ply'],
+}
+
+/**
+ * Compares the filetypes, taking into account different versions of the same
+ * filetype. For example, `.jpg` and `.jpeg` are considered the same filetype.
+ * @param extension - the extension of the file.
+ * @param filetype - the type of the file.
+ */
+const isFiletypeMatching = (extension: string, filetype?: string) => {
+  if (filetype === extension) {
+    return true
+  }
+  return (
+    filetype &&
+    ['jpg', 'jpeg'].includes(filetype) &&
+    ['jpg', 'jpeg'].includes(extension)
```

Comment:
Since there is the possibility to have many cases of an extensions having multiple forms, would a regex based elimination be more suited at this?

A simple regex to match extensions would look something like this: `.*\.jpe?g$`. This covers both considerations
- extension being the last part after the dot
- jpg or jpeg both being captured identically